misc/custom-script-shim

```diff
@@ -50,6 +50,19 @@ write_status() {
 	fi
 }
 
+check_binary_write_lock() {
+    set +e # disable exit on non-zero return code
+    lsof_output="$(lsof ${bin})"
+    if [ "$?" -eq 0 ]; then
+        echo "${HANDLER_BIN} is open by the following processes: "
+        echo "${lsof_output}"
+        echo "attempting to kill processes with open file handles to ${HANDLER_BIN}"
+        # suppress output and errors in case process with file-handle is already dead and kill gets called without a process id
+        lsof -t ${bin} | xargs kill -9 > /dev/null 2>&1
```

Comment:
Given that this appears to be a race condition, I'm not sure we should do anything but log and add a bit of retry/wait logic here. There was speculation that waagent may be holding up the file - we don't want to kill that or any other critical process.

Reply:
For now, let's hold off on killing any processes and see if the retries fix the problem. With this change, we'll also gather data about which processes are the culprits.